app/controllers/infra_networking_controller.rb

```diff
@@ -349,15 +350,21 @@ def replace_right_cell(options = {})
     @explorer = true
     @sb[:action] = action unless action.nil?
     if @sb[:action] || params[:display]
-      partial, _, @right_cell_text = set_right_cell_vars # Set partial name, action and cell header
+      # Set partial name, action and cell header
+      partial, action_type, @right_cell_text = set_right_cell_vars
     end
 
     if params[:action] == 'x_button' && params[:pressed] == 'infra_networking_tag'
       tagging
       return
     end
 
-    return if @in_a_form
+    if action_type == "dialog_form_button_pressed"
+      presenter = set_custom_button_dialog_presenter
+      render :json => presenter.for_render
+    end
+
+    return if @in_a_form && action
```

Comment:
Why `action` here? 

Looks like it comes from `options`... but I'm not seeing any changes to options in this pr..

(Should there be a `return` in that new `if` instead?)